File: languages/cpp/basics/03_condicionais_lacos.cpp
/*
Objetivo: praticar fluxo de controle em C++.
Como executar:
  powershell -ExecutionPolicy Bypass -File .\scripts\build-cpp-module.ps1 -Source .\languages\cpp\basics\03_condicionais_lacos.cpp
Conceitos treinados: if, else, for, range-based for.
Próximo exercício: 04_funcoes_referencias.cpp
*/

#include <iostream>
#include <vector>

int main() {
    int idade = 22;
    if (idade >= 18) {
        std::cout << "Maior de idade\n";
    } else {
        std::cout << "Menor de idade\n";
    }

    std::vector<int> valores{1, 2, 3, 4, 5};
    for (int valor : valores) {
        std::cout << valor << ' ';
    }
    std::cout << '\n';
    return 0;
}

File: languages/fortran/basics/03_condicionais_lacos.f90
! Objetivo: praticar if e do.
! Como executar:
!   powershell -ExecutionPolicy Bypass -File .\scriptsuild-fortran-module.ps1 -Source .\languagesortranasics_condicionais_lacos.f90
! Conceitos treinados: if, else, do.
! Próximo exercício: 04_subrotinas_funcoes.f90

program condicionais_lacos
    implicit none
    integer :: i, numero

    numero = 7

    if (mod(numero, 2) == 0) then
        print *, numero, "eh par"
    else
        print *, numero, "eh impar"
    end if

    do i = 1, 5
        print *, "i =", i
    end do
end program condicionais_lacos
